test/Microsoft.NET.Build.Tests/GivenThatWeWantToVerifyProjectReferenceCompat.cs

```diff
@@ -0,0 +1,213 @@
+        [InlineData("netcoreapp2.0", "netstandard2.0", true)]
+        [InlineData("netcoreapp2.0", "netcoreapp1.0", true)]
+        [InlineData("netcoreapp2.0", "netcoreapp1.1", true)]
+        [InlineData("netcoreapp2.0", "netcoreapp2.0", true)]
+
+        public void It_checks_for_valid_project_reference_compat(string referencerTarget, string dependencyTarget, bool buildSucceeds)
+        {
+            string identifier = "_" + referencerTarget.ToString() + "_" + dependencyTarget.ToString();
+            //  MSBuild isn't happy with semicolons in the path when doing file exists checks
+            identifier = identifier.Replace(';', '_');
+
+            TestProject referencerProject = GetTestProject("Referencer", referencerTarget, true);
+            TestProject dependencyProject = GetTestProject("Dependency", dependencyTarget, true);
+            referencerProject.ReferencedProjects.Add(dependencyProject);
+
+            //  Set the referencer project as an Exe unless it targets .NET Standard
+            if (!referencerProject.ShortTargetFrameworkIdentifiers.Contains("netstandard"))
+            {
+                referencerProject.IsExe = true;
+            }
+
+            //  Skip running test if not running on Windows
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (!referencerProject.BuildsOnNonWindows || !dependencyProject.BuildsOnNonWindows)
+                {
+                    return;
+                }
+            }
+
+            var testAsset = _testAssetsManager.CreateTestProject(referencerProject, nameof(It_checks_for_valid_project_reference_compat), identifier);
+            var restoreCommand = testAsset.GetRestoreCommand(relativePath: "Referencer").Execute().Should().Pass();
+            var appProjectDirectory = Path.Combine(testAsset.TestRoot, "Referencer");
+
+            var buildCommand = new BuildCommand(Stage0MSBuild, appProjectDirectory);
+            if (!buildSucceeds)
+            {
+                buildCommand = buildCommand.CaptureStdOut();
+            }
+
+            var result = buildCommand.Execute();
+
+            if (buildSucceeds)
+            {
+                result.Should().Pass();
+            }
+            else
+            {
+                result.Should().Fail()
+                    .And.HaveStdOutContaining("It cannot be referenced by a project that targets");
+            }
+        }
+
+        TestProject GetTestProject(string name, string target, bool isSdkProject)
+        {
+            TestProject ret = new TestProject()
+            {
+                Name = name,
+                IsSdkProject = isSdkProject
+            };
+
+            if (isSdkProject)
+            {
+                ret.TargetFrameworks = target;
+                //  Workaround for .NET Core 2.0
+                if (target.Equals("netcoreapp2.0", StringComparison.OrdinalIgnoreCase) || target.Equals("net462", StringComparison.OrdinalIgnoreCase))
```

Comment:
OIC; I will update the comment - "2.0.0-beta-001689-00" does indeed have an impact on net462; I think we should not need to set the RuntimeframeworkVersion for any of these.